chia/wallet/wallet_node.py

```diff
@@ -1129,18 +1130,15 @@ async def new_peak_wallet(self, new_peak: wallet_protocol.NewPeakWallet, peer: W
 
     async def new_peak_from_trusted(self, new_peak_hb: HeaderBlock, latest_timestamp: uint64, peer: WSChiaConnection):
         current_height: uint32 = await self.wallet_state_manager.blockchain.get_finished_sync_up_to()
```

Comment:
you end up calling this `async` function twice now. Once here and once inside `set_sync_mode()`. Since it's `async`, it might be a good idea to just call it once.

You could yield this value in the context manager for instance, to make it available to the caller

Reply:
I wanted to keep it simple and extend the sync context in a later PR because i think there is a bunch of stuff we could move there. But i added the height yielding here fbbd632dec5d148c2717c18869033ca632abbb24 :)